lib/zabbix/zabbix.go

```diff
@@ -203,3 +203,53 @@ func (a *API) GetTriggersInfo() (TriggerResults, error) {
 	}
 	return r.TriggerResults, nil
 }
+
+// AcknowledgeEvent acknowledges a single event and leaves a message.
+// It returns the event ID and an error if any.
+func (a *API) AcknowledgeEvent(eventID, message string) (int, error) {
+	params := struct {
+		EventIDs string `json:"eventids"`
+		// Event update action(s). This is bitmask field, any combination of values are acceptable.
+		// Possible values:
+		// 1 - close problem;
+		// 2 - acknowledge event;
+		// 4 - add message;
+		// 8 - change severity.
+		Action  int    `json:"action"`
+		Message string `json:"message"`
+	}{
+		EventIDs: eventID,
+		Action:   6,
+		Message:  message,
+	}
+
+	payload := a.BuildPayload(params, "event.acknowledge")
+
+	resp, err := utils.MakeZabbixRequest(http.MethodPost, a.URL, payload)
+
+	if err != nil {
+		return 0, fmt.Errorf("cannot make API request. error: %v", err)
+	}
+
+	var r struct {
+		Event map[string][]int `json:"result"`
+		Err   *apiError        `json:"error"`
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&r)
+	if err != nil {
+		return 0, fmt.Errorf("cannot decode response. error: %v", err)
+	}
+
+	//  Check if the response contains an error
+	if r.Err != nil {
+		return 0, fmt.Errorf("%v %v", r.Err.Message, r.Err.Data)
+	}
+
+	IDs := r.Event["eventids"]
+	if len(eventID) == 0 {
+		return 0, fmt.Errorf("no events found")
+	}
+
+	return IDs[0], nil
```

Comment:
I would want to suggest that this function just return the list of ids that were acknowledged and not bother with the len. That way it can easily be used to acknowledge multiple event (which is it's currently params definition on L211) and also defer the decision on the expected id len to the calling func.